internal/core/algorithm/ngt/ngt_test.go

```diff
@@ -0,0 +1,4537 @@
+						idxPath:             idxPath,
+						radius:              DefaultRadius,
+						epsilon:             DefaultEpsilon,
+						poolSize:            DefaultPoolSize,
+						bulkInsertChunkSize: 100,
+						objectType:          Uint8,
+						mu:                  &sync.RWMutex{},
+					},
+				},
+				checkFunc: func(w want, n NGT, e error) error {
+					if err := defaultCheckFunc(w, n, e); err != nil {
+						return err
+					}
+
+					// check inserted vector exists
+					v, err := n.GetVector(1)
+					if err != nil {
+						return err
+					}
+					if !reflect.DeepEqual(v, vec) {
+						t.Errorf("got vec is not the same as inserted vec, got: %v, want: %v", v, vec)
+					}
+
+					// check inserted vector can be searched
+					vs, err := n.Search(vec, 10, 0, 0)
+					if err != nil {
+						return err
+					}
+					if len(vs) != 1 || vs[0].ID != 1 || vs[0].Distance != 0 {
+						t.Errorf("got vec is not the same as inserted vec, got: %v, want: %v", vs, vec)
+					}
+
+					return nil
+				},
+			}
+		}(),
+		{
+			name: "return error when option return error",
+			args: args{
+				opts: []Option{
+					WithDimension(1),
+				},
+			},
+			want: want{
+				err: errors.NewErrCriticalOption("dimension", 1, errors.ErrInvalidDimensionSize(1, ngtVectorDimensionSizeLimit)),
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		test := tc
+		t.Run(test.name, func(tt *testing.T) {
+			tt.Parallel()
+			defer goleak.VerifyNone(tt, goleak.IgnoreCurrent())
+			if test.beforeFunc != nil {
+				test.beforeFunc(tt, test.args)
+			}
+			if test.afterFunc == nil {
+				test.afterFunc = defaultAfterFunc
+			}
+			if test.checkFunc == nil {
+				test.checkFunc = defaultCheckFunc
+			}
+
+			got, err := gen(test.args.isLoad, test.args.opts...)
+			defer func() {
+				if err := test.afterFunc(tt, got); err != nil {
+					tt.Error(err)
+				}
+			}()
+			if err := test.checkFunc(test.want, got, err); err != nil {
+				tt.Errorf("error = %v", err)
+			}
+		})
+	}
+}
+
+func Test_ngt_setup(t *testing.T) {
+	type fields struct {
+		idxPath             string
+		inMemory            bool
+		bulkInsertChunkSize int
+		objectType          objectType
+		radius              float32
+		epsilon             float32
+		poolSize            uint32
+		mu                  *sync.RWMutex
+	}
+	type want struct {
+		err error
+	}
+	type test struct {
+		name       string
+		fields     fields
+		want       want
+		checkFunc  func(want, error) error
+		beforeFunc func()
+		afterFunc  func(*testing.T, NGT) error
+	}
+	defaultCheckFunc := func(w want, err error) error {
+		if !errors.Is(err, w.err) {
+			return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
+		}
+		return nil
+	}
+	tests := []test{
+
+		{
+			name: "return nil when object type is uint8",
+			fields: fields{
+				objectType: Uint8,
+			},
+			want: want{},
+		},
+		{
+			name: "return nil when object type is float",
+			fields: fields{
+				objectType: Float,
+			},
+			want: want{},
+		},
```

Comment:
Should we add other test case like objectType is invalid?

Reply:
added test.